### PR TITLE
Implement 2048 game

### PR DESCRIPTION
**Pull request by olafchuszno**
[DEMO LINK](https://olafchuszno.github.io/js_2048_game/)

### CODE REVIEW

**Review by yevhenii-pyl — APPROVED** [review]
Solid: well structured and easy to follow.  🥇 

However, this code may benefit from comments. You can use this assignment as a good practise on how to comment your code when 'it all makes sense', but still the logic is pretty complicated and it may be hard to follow after some time.

**Comment by olafchuszno** [comment]
I always comment! I noticed that on every other task I've done I was asked to remove them, so I just assumed the same here and removed them before pushing.
